### DISCUSSION TITLE
i18n lookup function

### DISCUSSION THREAD

**Discussion by sbattou** [fnando/i18n-js#641] [Q&A]
Hey!

I'm currently using your library and it has been working great so far. 

I was going through some documentation and noticed that there is a `i18n.lookup('translation_key')` function that works but unfortunately it is not defined in the types/i18n-js index.d.ts.

Is there anyway you can add it or is there a better way to lookup whether a translation exists or not?

Thank you :)

**Comment by fnando** [comment]
Hey! Which version are you using?

**Reply by sbattou** [reply]
Hi @fnando!

Thank you for getting back to me. I'm using i18n-js version 3.8.0 and @types/i18n-js version 3.0.3

**Reply by fnando** [reply]
Got it. We don't maintain @types/i18n-js, so you should try to ping whoever is the author of that.

The latest version `i18n@next` doesn't expose lookup at all, so my approach would be using the `defaultValue`. Something like `i18n.t("missing", {defaultValue: ""})` which is a falsy value that can be checked like `if (!translation) {}`. If your translations may contain empty strings, replace that with `"__MISSING__"` or something, and extract it into a helper.

Now, just so I understand and see if we should have something exposed on the latest version… what's the motivation behind looking up translations to know whether they're set or not?

**Reply by sbattou** [reply]
Sounds good! `i18n.t("missing", {defaultValue: ""})` was actually my current work around so I will continue using it until the new version drops.

One of the use cases that I ran into recently is I have many objects and some require an additional accessibility label and some don't. So I need to check whether that object contains that key to add the accessibility label.

**Reply by sbattou** [reply]
Let me know if my use case makes sense or if I should go on about it differently :)

**Reply by fnando** [reply]
Ok, I think I'm gonna add `I18n#get(scope)`, which will return whatever it finds as it is, taking fallback locales into consideration.

**Reply by fnando** [reply]
Or, better putting, making the existing `I18n#get(scope)` function public.

**Reply by sbattou** [reply]
I'm assuming this function wouldn't return the missing message when a key doesn't exist?
